bin/tests/retries.rs

```diff
@@ -36,9 +35,12 @@ async fn test_retry_after_timeout() {
             *counter += 1;
             if *counter < attempts {
                 // Sleep enough time to mark the request as timed out by the client
-                thread::sleep(Duration::from_millis(timeout + 20));
+                return Some(Box::pin(tokio::time::sleep(Duration::from_millis(
```

Comment:
returning a future allows us to use tokio to sleep the task without blocking the integration test main thread